HCore-Storage/Client/Impl/GoogleCloudStorageClientImpl.cs

```diff
@@ -299,17 +299,31 @@ public async Task<string> GetSignedDownloadUrlAsync(string containerName, string
                 .CreateScoped(new string[] { "https://www.googleapis.com/auth/devstorage.read_only" })
                 .UnderlyingCredential as ServiceAccountCredential;
 
-            var urlSigner = UrlSigner.FromServiceAccountCredential(credential);
+            var urlSigner = UrlSigner.FromCredential(credential);
 
-            string signedUrl = await urlSigner.SignAsync(containerName, fileName, validityTimeSpan, HttpMethod.Get).ConfigureAwait(false);
+            var requestTemplate = UrlSigner.RequestTemplate.FromBucket(containerName)
+                .WithObjectName(fileName)
+                .WithHttpMethod(HttpMethod.Get);
+
+            var requestOptions = UrlSigner.Options.FromDuration(validityTimeSpan);
 
             if (!string.IsNullOrEmpty(downloadFileName))
             {
-                var contentDispositionHeader = new ContentDisposition() { FileName = downloadFileName };
+                var contentDispositionHeader = new ContentDisposition()
+                {
+                    FileName = downloadFileName
+                };
 
-                signedUrl = $"{signedUrl}&response-content-disposition={HttpUtility.UrlEncode(contentDispositionHeader.ToString())}";
+                var queryParameters = new Dictionary<string, IEnumerable<string>>
+                {
+                    { "response-content-disposition", [contentDispositionHeader.ToString()] }
+                };
+
+                requestTemplate = requestTemplate.WithQueryParameters(queryParameters);
             }
 
+            string signedUrl = await urlSigner.SignAsync(requestTemplate, requestOptions).ConfigureAwait(false);
+
             return signedUrl;
```

Comment:
The issue with the signed URL was that `response-content-disposition` was not part of the signed payload
The fix is to include it into `requestTemplate` before we call `SignAsync`
Only UrlSigner.RequestTemplate allows query parameters. We cannot do that with the other overloads

P.S. I had to give roles/owner permission
```
gcloud projects add-iam-policy-binding ninth-camera-414215 --member serviceAccount:test-sa@ninth-camera-414215.iam.gserviceaccount.com --role roles/owner
```
I tried with `roles/storage.legacyBucketOwner`, but it no longer works - `INVALID_ARGUMENT: Role roles/storage.legacyBucketOwner is not supported for this resource.`
Here is a link to the IAM roles page - [click](https://cloud.google.com/storage/docs/access-control/iam-roles)

Reply:
I tested all the other implemented methods and they still work correctly when I gave the roles/role to owner

Reply:
Right now the constructor looks like this
```C#

public GoogleCloudStorageClientImpl(string connectionString)
{
	int firstIndex = connectionString.IndexOf(":");
	if (firstIndex == -1)
		throw new Exception("Google Cloud connection string is invalid");
	_projectId = connectionString.Substring(0, firstIndex);
	if (string.IsNullOrEmpty(_projectId))
		throw new Exception("The Google Cloud project ID is invalid");
	_credentialsJson = connectionString.Substring(firstIndex + 1);
	if (string.IsNullOrEmpty(_credentialsJson))
		throw new Exception("The Google Cloud credentials JSON is invalid");
}
```
Unless we have used this in the past with a different JSON format than the one provided by the Google JSON service account, I don't see how this works.
Should I change it, so that it parses the `_projectId` and `_credentialsJson` correctly?

P.S. I can not comment on code I have not touched.